### PR TITLE
[luci-interpreter] Introduce PAL kernel wrappers

### PR DESCRIPTION
**Pull request by binarman**
This commit introduces wrappers for computation code to abstract it
from platform specific features, like system libraries.

related issue: #7288
draft: #7336 

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@partner.samsung.com>

### CODE REVIEW

**Review by BalyshevArtem — DISMISSED** [review]

File: compiler/luci-interpreter/pal/linux/PALBatchToSpaceND.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_PAL_BATCHTOSPACEND_H
+#define LUCI_INTERPRETER_PAL_ARGMAX_H
```
Comment (binarman) [inline compiler/luci-interpreter/pal/linux/PALBatchToSpaceND.h:18]:
```suggestion
#define LUCI_INTERPRETER_PAL_BATCHTOSPACEND_H
```

File: compiler/luci-interpreter/pal/linux/PALDepthToSpace.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_PAL_DEPTHTOSPACE_H
+#define LUCI_INTERPRETER_PAL_ARGMAX_H
```
Comment (binarman) [inline compiler/luci-interpreter/pal/linux/PALDepthToSpace.h:18]:
```suggestion
#define LUCI_INTERPRETER_PAL_DEPTHTOSPACE_H
```

File: compiler/luci-interpreter/pal/mcu/PALDepthToSpace.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_PAL_DEPTHTOSPACE_H
+#define LUCI_INTERPRETER_PAL_ARGMAX_H
```
Comment (binarman) [inline compiler/luci-interpreter/pal/mcu/PALDepthToSpace.h:18]:
```suggestion
#define LUCI_INTERPRETER_PAL_DEPTHTOSPACE_H
```

File: compiler/luci-interpreter/pal/mcu/PALSoftmax.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  params->diff_min =
+    -tflite::CalculateInputRadius(kScaledDiffIntegerBits, params->input_left_shift);
+}
+
+template <typename T>
+static inline void Softmax(const tflite::SoftmaxParams &params,
+                           const tflite::RuntimeShape &input_shape, const T *input_data,
+                           const tflite::RuntimeShape &output_shape, T *output_data)
+{
+  // MARK: At this moment this operation doesn't support on mcu
```
Comment (binarman) [inline compiler/luci-interpreter/pal/mcu/PALSoftmax.h:52]: I think it is better to add assert, so we will not forgot about this
```suggestion
  // MARK: At this moment this operation doesn't support on mcu
  assert(false && "Softmax NYI");
```

**Review by SlavikMIPT — APPROVED** [review]
LGTM

**Review by BalyshevArtem — APPROVED** [review]

**Pull request merged**